### PR TITLE
Make text about render :html option clearer to state, text will get e…

### LINKED ISSUE #22648: Layouts and Rendering documentation section 2.2.8 Rendering HTML creates ambiguity around use of `html_safe`
The [2.2.8 Rendering HTML][0] section of the [layouts and rendering documentation][1] contains a note about `html_safe` that says "This option will escape HTML entities if the string is not HTML safe." This is true when using render, but quite the [opposite][2] when using templates, the latter being the most common use of `html_safe`.

It would probably best to include "When using render" in the note since the note itself stands out very much from the rest of the context and someone loading the page and searching for "html_safe" will very easily see the note and not necessarily appreciate the context.

Anecdotally, the layouts and rendering page is the 2nd result when I Google search for ["rails guide view"][3] and is the first result that mentions `html_safe`, so I think it's not unreasonable for new people to stumble upon it when trying to learn about views.

I also think it would probably be a good idea to rename one of the functions as they currently mean the opposite, the render says "please escape" and the template says "please *don't* escape". I think it's use in templates is more confusing, as it's describing the state of the parameter (the parameter is html safe) rather than an action being applied to it, so something like `html_no_escape` or `html_dont_escape` more clearly indicates escaping will *NOT* occur. Alternatively, the render function could be renamed to something like `html_escape`.

[0]: http://guides.rubyonrails.org/layouts_and_rendering.html#rendering-html
[1]: http://guides.rubyonrails.org/layouts_and_rendering.html
[2]: http://guides.rubyonrails.org/active_support_core_extensions.html#output-safety
[3]: https://www.google.com/search?q=rails+guide+view

### PR DESCRIPTION
**Pull request by vipulnsward**
…scaped if not marked as safe by using html_safe

Fixes #22648

### CODE REVIEW

File: guides/source/layouts_and_rendering.md [hunk 1]
```diff
@@ -238,7 +238,7 @@ TIP: This is useful when you're rendering a small snippet of HTML code.
 However, you might want to consider moving it to a template file if the markup
 is complex.
 
-NOTE: This option will escape HTML entities if the string is not HTML safe.
+NOTE: When using `html:` option, HTML entities will be escaped if the string is not marked as HTML safe by using `html_safe` method.
```
Comment (prathamesh-sonpatki) [inline guides/source/layouts_and_rendering.md]: We are `explaining `html` option only, so no need to mention it explicitly.

Comment (vipulnsward) [inline guides/source/layouts_and_rendering.md]: Following other places, to make the note explicit.

**Comment by prathamesh-sonpatki** [comment]
I think this section is about explaining how `render: html` works rather than `html_safe`. 

To resolve this confusion - 

we can simply remove `.html_safe` from `render html: "<strong>Not Found</strong>" ` and 

add `HTML entities will be escaped if the string is not marked as HTML safe by using `html_safe` method.` in the `NOTE`.

**Comment by vipulnsward** [comment]
@prathamesh-sonpatki actually that will be confusing. We are trying to show how we can use :html option to actually render html with tags. The note is just to warn when it won't render html as is.

**Comment by prathamesh-sonpatki** [comment]
:+1:

File: guides/source/layouts_and_rendering.md (hunk 1 above)
Comment (kaspth) [inline guides/source/layouts_and_rendering.md]: `html_safe` is still officially considered private and one people shouldn't call as far as I know.

cc @rafaelfranca

Comment (vipulnsward) [inline guides/source/layouts_and_rendering.md]: true, I believe we can replace with raw?

Comment (kaspth) [inline guides/source/layouts_and_rendering.md]: I'd prefer that, yes :)

Comment (vipulnsward) [inline guides/source/layouts_and_rendering.md]: Although its hard to determine what needs to be replaced. `html_safe` has its own section defined https://github.com/rails/rails/blob/3d1f95d0e8ca70f5d892d9b2ed78e99ff7cb3bde/guides/source/active_support_core_extensions.md#safe-strings with output_safety.

So only replace uses of `html_safe` in `AV` ?

Comment (rafaelfranca) [inline guides/source/layouts_and_rendering.md]: > So only replace uses of html_safe in AV ?

Yes, that.

Comment (prathamesh-sonpatki) [inline guides/source/layouts_and_rendering.md]: This example should also be changed?

**Pull request merged**